x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -752,12 +786,13 @@ private static void addMetricFieldMapping(final XContentBuilder builder, final S
                 builder.field(fieldProperty, fieldProperties.get(fieldProperty));
             }
         } else {
-            final String[] supportedAggsArray = metricType.supportedAggs();
-            // We choose max as the default metric
-            final String defaultMetric = List.of(supportedAggsArray).contains("max") ? "max" : supportedAggsArray[0];
+            var supported = getSupportedMetrics(metricType, fieldProperties);
+
+            String[] supportedMetricsArray = supported.supportedMetrics.toArray(String[]::new);
+
             builder.field("type", AggregateMetricDoubleFieldMapper.CONTENT_TYPE)
-                .array(AggregateMetricDoubleFieldMapper.Names.METRICS, supportedAggsArray)
-                .field(AggregateMetricDoubleFieldMapper.Names.DEFAULT_METRIC, defaultMetric)
+                .array(AggregateMetricDoubleFieldMapper.Names.METRICS, supportedMetricsArray)
```

Comment:
nit: looks like you could use `stringListField` and avoid the array

Reply:
Ah, I thought there might be an API like that, but I couldn't find it. Thanks!